plc4j/utils/opm/src/main/java/org/apache/plc4x/java/opm/PlcEntityManager.java

```diff
@@ -0,0 +1,380 @@
+    //
+    //------------------------------------------------------------------------------------------------
+
+    /**
+     * Basic Intersector for all methods on the proxy object.
+     * It checks if the invoked method is a getter and if so, only retrieves the requested field, forwarding to
+     * the {@link #fetchValueForGetter(Object, Method)} method.
+     * <p>
+     * If the field is no getter, then all fields are refreshed by calling {@link #refetchAllFields(Object)}
+     * and then, the method is invoked.
+     *
+     * @param o    Object to intercept
+     * @param m    Method that was intercepted
+     * @param c    Callable to call the method after fetching the values
+     * @param that Reference to the proxy object
+     * @return possible result of the original methods invocation
+     * @throws OPMException Problems with plc / proxying
+     */
+    @RuntimeType
+    // TODO: avoid using single letter params.
+    public Object intercept(@This Object o, @Origin Method m, @SuperCall Callable<?> c, @Super Object that) throws OPMException {
+        LOGGER.trace("Invoked method {} on connected PlcEntity {}", m.getName(), that);
+
+        if (m.getName().startsWith("get")) {
+            // Fetch single value
+            LOGGER.trace("Invoked method {} is getter, trying to find annotated field and return requested value",
+                m.getName());
+            return fetchValueForGetter(that, m);
+        }
+
+        if (m.getName().startsWith("is") && (m.getReturnType() == boolean.class || m.getReturnType() == Boolean.class)) {
+            // Fetch single value
+            LOGGER.trace("Invoked method {} is boolean flag method, trying to find annotated field and return requested value",
+                m.getName());
+            return fetchValueForGetter(that, m);
+        }
+
+        // Fetch all values, than invoke method
+        try {
+            LOGGER.trace("Invoked method is no getter, refetch all fields and invoke method {} then", m.getName());
+            refetchAllFields(o);
+            return c.call();
+        } catch (Exception e) {
+            throw new OPMException("Unbale to forward invokation " + m.getName() + " on connected PlcEntity", e);
+        }
+    }
+
+    /**
+     * Renews all values of all Fields that are annotated with {@link PlcEntity}.
+     *
+     * @param o Object to refresh he fields on.
+     * @throws OPMException
+     */
+    private void refetchAllFields(Object o) throws OPMException {
+        Class<?> superclass = o.getClass().getSuperclass();
+        PlcEntity plcEntity = superclass.getAnnotation(PlcEntity.class);
+
+        try (PlcConnection connection = driverManager.getConnection(plcEntity.value())) {
+            // Catch the exception, if no reader present (see below)
+            PlcReader plcReader = connection.getReader().get();
+
+            // Build the query
+            PlcReadRequest.Builder builder = plcReader.readRequestBuilder();
```

Comment:
so currently only read is possible?

Reply:
Yes, perhaps I didnt state this enough. Currently, only read. It should not be that big of a deal to add writing. I'll open a Jira for that.

Reply:
Done.. is PLC4X-70